src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -699,19 +700,26 @@ private String encodeMdcFields(AuthorizationRequestData authorizationData) {
 
     private Either<RedirectConfigurationException, URI> getRedirectUrlForPsp(
                                                                              String pspId,
-                                                                             String paymentTypeCode
+                                                                             String paymentTypeCode,
+                                                                             String channelCode
     ) {
-        String urlKey = "%s-%s".formatted(pspId, paymentTypeCode);
-        if (redirectBeApiCallUriMap.containsKey(urlKey)) {
-            return Either.right(redirectBeApiCallUriMap.get(urlKey));
-        } else {
-            return Either.left(
-                    new RedirectConfigurationException(
-                            "Missing key for redirect return url with key: [%s]".formatted(urlKey),
-                            RedirectConfigurationType.BACKEND_URLS
-                    )
-            );
+
+        return searchRedirectUrlForPsp(channelCode, pspId, paymentTypeCode);
```

Comment:
Add junit tests to cover all the search cases: stop at first iteration, second, third and no match

Reply:
Done.